src/model/markerscollection.js

```diff
@@ -0,0 +1,190 @@
+		}
+
+		if ( !( liveRange instanceof LiveRange ) ) {
+			/**
+			 * Added range is not an instance of LiveRange.
+			 *
+			 * @error markers-collection-add-range-not-live-range
+			 */
+			throw new CKEditorError( 'markers-collection-add-range-not-live-range: Added range is not an instance of LiveRange.' );
+		}
+
+		this._nameToRange.set( markerName, liveRange );
+		this.fire( 'add', markerName, Range.createFromRange( liveRange ) );
+	}
+
+	/**
+	 * Returns the live range that was added to `MarkersCollection` under given `markerName`.
+	 *
+	 * @param {String} markerName Name of range to get.
+	 * @returns {module:engine/model/liverange~LiveRange|null} Range added to collection under given name or `null` if
+	 * no range was added with that name.
+	 */
+	get( markerName ) {
+		return this._nameToRange.get( markerName ) || null;
+	}
+
+	/**
+	 * Removes a live range from markers collection.
+	 *
+	 * @param {String|module:engine/model/liverange~LiveRange} nameOrRange Name of live range to remove or live range instance.
+	 * @returns {Boolean} `true` is passed if range was found and removed from the markers collection, `false` otherwise.
+	 */
+	remove( nameOrRange ) {
+		if ( nameOrRange instanceof LiveRange ) {
+			const name = this._getMarkerName( nameOrRange );
+
+			if ( name ) {
+				return this.remove( name );
+			}
+
+			return false;
+		}
+
+		const range = this._nameToRange.get( nameOrRange );
+
+		if ( range ) {
+			this._nameToRange.delete( nameOrRange );
+			this.fire( 'remove', nameOrRange, Range.createFromRange( range ) );
+
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Substitutes given `oldLiveRange`, that was already added to the markers collection, with given `newLiveRange`.
+	 *
+	 * This method is basically a wrapper for using {@link module:engine/model/markerscollection~MarkersCollection#removeRange removeRange}
+	 * followed by using {@link module:engine/model/markerscollection~MarkersCollection#addRange addRange}.
+	 *
+	 * **Note**: this method does not change properties of `oldLiveRange`.
+	 *
+	 * @param {String|module:engine/model/liverange~LiveRange} nameOrRange Name of range or range instance to be changed.
+	 * @param {module:engine/model/liverange~LiveRange} newLiveRange Live range to be added.
+	 * @returns {Boolean} `true` if `oldLiveRange` was found and changed, `false` otherwise.
+	 */
+	update( nameOrRange, newLiveRange ) {
+		if ( nameOrRange instanceof LiveRange ) {
+			const name = this._getMarkerName( nameOrRange );
+
+			if ( name ) {
+				return this.update( name, newLiveRange );
+			}
+
+			return false;
+		}
+
+		const removed = this.remove( nameOrRange );
+
+		if ( removed ) {
+			this.add( nameOrRange, newLiveRange );
+		}
+
+		return removed;
+	}
+
+	/**
+	 * Destroys markers collection.
+	 */
+	destroy() {
+		this.stopListening();
+	}
+
+	/**
+	 * Returns range for given name.
+	 *
+	 * @private
```

Comment:
Still, range instance is a terrible ID. Also, it is not a point to introduce more features that we need at this stage. Then, we need to maintain them, what is usually more costly them introducing them. If we are unsure we will need this feature we should not bring it.